File: src/test/fixtures/java-example/src/com/example/utils/Calculator.java
package com.example.utils;

public interface Calculator {
    double add(double a, double b);
    double subtract(double a, double b);
    double multiply(double a, double b);
    double divide(double a, double b) throws ArithmeticException;
}

class BasicCalculator implements Calculator {
    @Override
    public double add(double a, double b) {
        return a + b;
    }

    @Override
    public double subtract(double a, double b) {
        return a - b;
    }

    @Override
    public double multiply(double a, double b) {
        return a * b;
    }

    @Override
    public double divide(double a, double b) throws ArithmeticException {
        if (b == 0.0) {
            throw new ArithmeticException("Division by zero");
        }
        return a / b;
    }
}

public final class MathUtils {
    private MathUtils() {
        throw new UnsupportedOperationException("Utility class cannot be instantiated");
    }

    public static long factorial(int n) {
        if (n < 0) throw new IllegalArgumentException("Factorial is not defined for negative numbers");
        if (n <= 1) return 1;
        return n * factorial(n - 1);
    }

    public static boolean isPrime(int n) {
        if (n < 2) return false;
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) return false;
        }
        return true;
    }

    public static int fibonacci(int n) {
        if (n < 0) throw new IllegalArgumentException("Fibonacci is not defined for negative numbers");
        if (n == 0) return 0;
        if (n == 1) return 1;
        return fibonacci(n - 1) + fibonacci(n - 2);
    }

    public static <T extends Comparable<T>> T max(T a, T b) {
        return a.compareTo(b) >= 0 ? a : b;
    }

    public static <T extends Comparable<T>> T min(T a, T b) {
        return a.compareTo(b) <= 0 ? a : b;
    }

    public static void main(String[] args) {
        Calculator calc = new BasicCalculator();
        System.out.println("5 + 3 = " + calc.add(5.0, 3.0));
        System.out.println("10! = " + factorial(10));
        System.out.println("Fibonacci 8 = " + fibonacci(8));
        System.out.println("Is 17 prime? " + isPrime(17));
    }
}

File: src/test/fixtures/elixir-example/lib/calculator.ex
defmodule Example.Calculator do
  @moduledoc """
  Basic calculator operations
  """

  @spec add(number(), number()) :: number()
  def add(a, b) when is_number(a) and is_number(b), do: a + b

  @spec subtract(number(), number()) :: number()
  def subtract(a, b) when is_number(a) and is_number(b), do: a - b

  @spec multiply(number(), number()) :: number()
  def multiply(a, b) when is_number(a) and is_number(b), do: a * b

  @spec divide(number(), number()) :: {:ok, number()} | {:error, String.t()}
  def divide(_a, 0), do: {:error, "Division by zero"}
  def divide(a, b) when is_number(a) and is_number(b), do: {:ok, a / b}
end

defmodule Example.MathUtils do
  @moduledoc """
  Mathematical utility functions
  """

  @spec factorial(non_neg_integer()) :: pos_integer()
  def factorial(0), do: 1
  def factorial(n) when n > 0, do: n * factorial(n - 1)

  @spec is_prime?(integer()) :: boolean()
  def is_prime?(n) when n < 2, do: false
  def is_prime?(2), do: true
  def is_prime?(n) when rem(n, 2) == 0, do: false
  def is_prime?(n) do
    limit = :math.sqrt(n) |> trunc()
    not Enum.any?(3..limit//2, &(rem(n, &1) == 0))
  end

  @spec fibonacci(non_neg_integer()) :: non_neg_integer()
  def fibonacci(0), do: 0
  def fibonacci(1), do: 1
  def fibonacci(n) when n > 1, do: fibonacci(n - 1) + fibonacci(n - 2)
end